native/dlaqp2.go

```diff
@@ -0,0 +1,112 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dlaqp2 computes a QR factorization with column pivoting of the block A[offset:m, 0:n]
+// of the m×n matrix A. The block A[0:offset, 0:n] is accordingly pivoted, but not factorized.
+//
+// On exit, the upper triangle of block A[offset:m, 0:n] is the triangular factor obtained.
+// The elements in block A[offset:m, 0:n] below the diagonal, together with tau, represent
+// the orthogonal matrix Q as a product of elementary reflectors.
+//
+// offset is number of rows of the matrix A that must be pivoted but no factorized.
+// offset must not be negative otherwise Dlaqp2 will panic.
+//
+// On exit, if jpvt[i] = k, then the ith column of A*P was the kth column of A.
+// jpvt must have length n, otherwise Dlapq2 will panic.
+//
+// On exit tau holds the scalar factors of the elementary reflectors. It must have length
+// at least min(m, n) otherwise Dlapq2 will panic.
+//
+// vn1 and vn2 hold the partial and complete column norms respectively. They must have length n,
+// otherwise Dlapq2 will panic.
+//
+// work contains temporary storage and must have length m. Dlapq2 will panic if there is
```

Comment:
In my PRs I stopped writing that work is a temporary memory, and just write how long it must be.